Sources/System/Internals/WindowsSyscallAdapters.swift

```diff
@@ -105,4 +105,11 @@ internal func pwrite(
   return Int(nNumberOfBytesWritten)
 }
 
+@inline(__always)
+internal func ftruncate(
+  _ fd: Int32, _ length: off_t
+) -> Int32 {
+  _chsize(fd, numericCast(length))
+}
```

Comment:
Full disclosure: I have not tested the Windows path. AFAICT there isn't a Windows CI pipeline?

Reply:
I believe it's best to surround the addition of `resize` with `#if !os(Windows)`.

Generally, I don't think it's a good idea to try to shoehorn Windows support into POSIX code paths here, or anywhere else in this code base. Swift System is not intended to provide a portable API layer, and the Windows SDK is different enough that it deserves its own, standalone, implementations.

Reply:
OK, apologies for that, I was just following the pattern I saw with the existing `WindowsSyscallAdapters`. I added some fixups which move this API to only non-Windows platforms:

* 02daf4b `fixup: Move FileDescriptor.resize to extension with #if !os(Windows)`
* 7fab65f `fixup: Remove Windows syscall adaptor for ftruncate`